.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,26 +1,62 @@
 
-### Commits
+  - Fill out the template below to describe the changes contributed by the pull request. That will give reviewers the context they need to do the review.
+  
+  - Make sure that the change passes the automated tests. You can [build the entire project](https://github.com/apache/avro/blob/master/BUILD.md) or just the [language-specific SDK](https://avro.apache.org/project/how-to-contribute/#unit-tests).
 
-- [ ] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](https://chris.beams.io/posts/git-commit/)":
-  1. Subject is separated from body by a blank line
-  1. Subject is limited to 50 characters (not including Jira issue reference)
-  1. Subject does not end with a period
-  1. Subject uses the imperative mood ("add", not "adding")
-  1. Body wraps at 72 characters
-  1. Body explains "what" and "why", not "how"
+  - Each pull request should address only one issue, not mix up code from multiple issues.
+  
+  - Each commit in the pull request has a meaningful commit message (including the JIRA id)
 
-### Documentation
+  - Once all items of the checklist are addressed, remove the above text and this checklist, leaving only the filled out template below.
 
-- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
-  - All the public functions and the classes in the PR contain Javadoc that explain what it does
+  - Every commit message references Jira issues in their subject lines. In addition, commits follow the guidelines from [How to write a good git commit message](https://chris.beams.io/posts/git-commit/)
+    1. Subject is separated from body by a blank line
+    1. Subject is limited to 50 characters (not including Jira issue reference)
+    1. Subject does not end with a period
+    1. Subject uses the imperative mood ("add", not "adding")
+    1. Body wraps at 72 characters
+    1. Body explains "what" and "why", not "how"
+
+-->
+
+## What is the purpose of the change
+
+*(For example: This pull request improves file read performance by buffering data, fixing AVRO-XXXX.)*
+
+
+## Verifying this change
+
+*(Please pick one of the following options)*
+
+This change is a trivial rework / code cleanup without any test coverage.
+
+*(or)*
+
+This change is already covered by existing tests, such as *(please describe tests)*.
+
+*(or)*
+
+This change added tests and can be verified as follows:
+
+*(example:)*
+- *Extended interop tests to verify consistent valid schema names between SDKs*
+- *Added test that validates that Java throws an AvroRuntimeException on invalid binary data*
+- *Manually verified the change by building the website and checking the new redirect*
+
+
+## Documentation
+
+- Does this pull request introduce a new feature? (yes / no)
+- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
```

Comment:
Is there a generic way to say "Javadoc" that includes in-line code documentation like doxygen and python docs?